### ISSUE TITLE
BinaryExpression parens

### ISSUE THREAD

**Issue by cah4a** [rusty-ecma/RESW#7]
Hi!

First, thanks for the awesome crates!

I am making a transpiler for my new programming language and face a parenthesis issue.

For the follow expression:
```rust
BinaryExpr {
    operator: Times,
    left: Lit(
        Number(
            "1",
        ),
    ),
    right: Binary(
        BinaryExpr {
            operator: Plus,
            left: Lit(
                Number(
                    "2",
                ),
            ),
            right: Lit(
                Number(
                    "3",
                ),
            ),
        },
    ),
}
```

RESW output is:
```js
(1 * 2 + 3);
```

Which obviously parses back using precedence rules into:
```rust
BinaryExpr {
    operator: Plus,
    left: Binary(
        BinaryExpr {
            operator: Times,
            left: Lit(
                Number(
                    "1",
                ),
            ),
            right: Lit(
                Number(
                    "2",
                ),
            ),
        },
    ),
    right: Lit(
        Number(
            "3",
        ),
    ),
}
```

In that case, we are losing information about the expression's precedence.

**Comment by FreeMasen** [comment]
Closed by #10 

A new version was published as 0.5.0 on crates.io

Thanks again!

**Comment by cah4a** [comment]
Thanks for a quick reaction! It means a lot.
